File: dto-mapping-benchmark/src/main/java/com/example/dtomappingbenchmark/v4/controller/PostController.java
package com.example.dtomappingbenchmark.v4.controller;

import com.example.dtomappingbenchmark.v4.dto.PostDto;
import com.example.dtomappingbenchmark.v4.service.PostService;
import lombok.RequiredArgsConstructor;
import org.springframework.http.ResponseEntity;
import org.springframework.web.bind.annotation.PostMapping;
import org.springframework.web.bind.annotation.RequestBody;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RestController;

@RestController
@RequiredArgsConstructor
@RequestMapping("/api/v4/posts")
public class PostController {
    private final PostService postService;

    @PostMapping
    public ResponseEntity<?> create(@RequestBody PostDto dto){
        return ResponseEntity.ok(postService.create(dto));
    }
}

File: object-mapping-benchmark-v1/src/main/kotlin/yu/selab/cj/objectmappingbenchmarkv1/post/controller/PostController.kt
package yu.selab.cj.objectmappingbenchmarkv1.post.controller

import org.springframework.http.ResponseEntity
import org.springframework.web.bind.annotation.DeleteMapping
import org.springframework.web.bind.annotation.GetMapping
import org.springframework.web.bind.annotation.PathVariable
import org.springframework.web.bind.annotation.PostMapping
import org.springframework.web.bind.annotation.RequestBody
import org.springframework.web.bind.annotation.RequestMapping
import org.springframework.web.bind.annotation.RestController
import yu.selab.cj.objectmappingbenchmarkv1.post.command.CreatePostDto
import yu.selab.cj.objectmappingbenchmarkv1.post.service.PostService

@RestController
@RequestMapping("/posts")
class PostController(private val postService: PostService) {

    @GetMapping
    fun getPostList()
        = ResponseEntity.ok().body(postService.getPostList())

    @GetMapping("/{id}")
    fun getPost(@PathVariable id : Long)
        = ResponseEntity.ok().body(postService.getPost(id))

    @PostMapping
    fun createPost(@RequestBody dto : CreatePostDto)
        = ResponseEntity.ok().body(postService.createPost(dto))

    @DeleteMapping("/{id}")
    fun deletePost(@PathVariable id : Long)
        = ResponseEntity.ok().body(postService.deletePost(id))

}